### ISSUE TITLE
Failed to demangle function with template parameter

### ISSUE THREAD

**Issue by mitsuhiko** [gimli-rs/cpp_demangle#99]
I came across this symbol which does not demangle:

```
_ZL29SupportsTextureSampleCountMTLPU19objcproto9MTLDevice11objc_objectm
```

With an extra underscore c++filt does this:

```
SupportsTextureSampleCountMTL(id<MTLDevice>, unsigned long)
```

**Comment by mitsuhiko** [comment]
Found another one:

```
_ZL19StringContainsEmojiP8NSString
```

which is

```
StringContainsEmoji(NSString*)
```

These seem to come from a Objective-C++ codebase.

**Comment by fitzgen** [comment]
Thanks for the bug report!

**Comment by fitzgen** [comment]
This got fixed somewhere in the 0.2.7 release. Will add a regression test as well.
